python-checks/src/main/java/org/sonar/python/checks/CipherBlockChainingCheck.java

```diff
@@ -0,0 +1,162 @@
+      return;
+    }
+    checkAssignmentStatement(assignmentStatement, callExpression, "encryptor", ctx);
+  }
+
+  private static void checkPyCryptoCall(CallExpression callExpression, SubscriptionContext ctx) {
+    RegularArgument modeArgument = TreeUtils.nthArgumentOrKeyword(1, "mode", callExpression.arguments());
+    if (modeArgument == null) {
+      return;
+    }
+    Expression modeArgumentExpression = modeArgument.expression();
+    if (!(modeArgumentExpression instanceof HasSymbol)) {
+      return;
+    }
+    Symbol symbol = ((HasSymbol) modeArgumentExpression).symbol();
+    if (symbol == null || !"MODE_CBC".equals(symbol.name())) {
+      return;
+    }
+    RegularArgument ivArgument = TreeUtils.nthArgumentOrKeyword(2, "iv", callExpression.arguments());
+    if (ivArgument == null || !isStaticInitializationVector(ivArgument.expression())) {
+      return;
+    }
+    AssignmentStatement assignmentStatement = (AssignmentStatement) TreeUtils.firstAncestorOfKind(callExpression, Tree.Kind.ASSIGNMENT_STMT);
+    if (assignmentStatement == null || assignmentStatement.assignedValue() != callExpression) {
+      return;
+    }
+    checkAssignmentStatement(assignmentStatement, callExpression, "encrypt", ctx);
+  }
+
+  private static void checkAssignmentStatement(AssignmentStatement assignmentStatement, CallExpression callExpression, String suspiciousCallee, SubscriptionContext ctx) {
+    assignmentStatement.lhsExpressions().stream()
+      .filter(exprList -> exprList.expressions().size() == 1)
+      .flatMap(exprList -> exprList.expressions().stream())
+      .filter(expression -> expression.is(Tree.Kind.NAME))
+      .forEach(name -> {
+        Symbol symbol = ((Name) name).symbol();
+        if (symbol == null) {
+          return;
+        }
+        symbol.usages().stream()
+          .map(Usage::tree)
+          .filter(t -> isWithinCallTo(t, suspiciousCallee))
+          .findFirst()
+          .ifPresent(t -> ctx.addIssue(t, MESSAGE).secondary(callExpression, null));
+      });
+  }
+
+  private static boolean isWithinCallTo(Tree tree, String calleeName) {
+    CallExpression callExpression = (CallExpression) TreeUtils.firstAncestorOfKind(tree, Tree.Kind.CALL_EXPR);
+    if (callExpression == null) {
+      return false;
+    }
+    return callExpression.callee().is(Tree.Kind.QUALIFIED_EXPR) && ((QualifiedExpression) callExpression.callee()).name().name().equals(calleeName);
+  }
+
+  private static boolean isStaticInitializationVector(Expression expression) {
+    if (expression.is(Tree.Kind.CALL_EXPR) || TreeUtils.hasDescendant(expression, tree -> tree.is(Tree.Kind.CALL_EXPR))) {
+      return false;
+    }
+    if (expression.is(Tree.Kind.NAME)) {
+      Expression singleAssignedValue = Expressions.singleAssignedValue((Name) expression);
+      if (singleAssignedValue == null) {
+        return false;
+      }
+      return isStaticInitializationVector(singleAssignedValue);
```

Comment:
I think we can have a StackOverflowException in the following case:
```python
def soe():
    static_vector = other_static
    other_static = static_vector
    enc2 = AES.new(key, AES.MODE_CBC, static_vector)  # 2nd location
    cipher_text = enc2.encrypt(pad(data, AES.block_size))  # Noncompliant
```

Reply:
Indeed, thanks!